### ISSUE TITLE
WorldGuard region issue

### ISSUE THREAD

**Issue by DarkObscurity** [SkriptLang/Skript#2737]
### Description

Region related effect/conditions do not seem to work

### Steps to Reproduce

For example do a loop-all block in region "<region>" and it fails.

### Expected Behavior

Region related effects/conditions should work? 

### Errors / Screenshots

18:44:42.629 [CHAT] Can't understand this loop: 'loop all blocks in region "end-spawn-floor":

### Server Information
* **Server version/platform:**    Paper Spigot version git-paper-31 (1.15.1)    
* **Skript version:**       2.4.1         
World Guard version:  7.0.2-SNAPSHOT;1994-9adac4f

### Additional Context

UPDATE:

It looks like skript isn't comparing the region at the player to the requested region at all or something?

**Comment by APickledWalrus** [comment]
You can’t really put in strings for region, but you could do

```
set {_region} to region at (location within the region)
loop all blocks in region {_region}:
```

**Comment by DarkObscurity** [comment]
This has always worked? So why wouldn't it all of a sudden?.. @APickledWalrus

**Comment by Runakai1** [comment]
> This has always worked? So why wouldn't it all of a sudden?.. @APickledWalrus

Welcome to Skript

**Comment by DarkObscurity** [comment]
@Runakai1 Eh what? hahaha 

Common, this  has got to be a bug right? I've been using this throughout versions of skript starting on 1.12.2?

**Comment by APickledWalrus** [comment]
What version were you using before? There were some changes a bit ago I believe. It may be an issue with the expression though, so you could try putting the name in a variable and using that to see it it works? Don’t forget to include the world in the region name though, for example, “end-spawn-floor in world_the_end”.

According to the docs though, using strings is not supported for all region plugins.

**Comment by DarkObscurity** [comment]
@APickledWalrus Thank you very much! It works! I put the region name in a variable and Skript loads it! Althought i'm not clearly understanding the issue here because putting a string in a variable to make skript parse that variable as a string, to me, still sounds like skript is parsing it as a string xD

**Comment by DarkObscurity** [comment]
@APickledWalrus Nevermind... it still does not work... Skript can parse it but it doesn't do anything.... like for example parsing "player is in region'' doesn't do anything when the player is in the region defined by the variable...

**Comment by APickledWalrus** [comment]
What all changed when using strings stopped working? I believe you aren’t able to use strings though, but my original example in my first comment should work.

**Comment by DarkObscurity** [comment]
I think I was using version 2.4-beta7 @APickledWalrus

**Comment by DarkObscurity** [comment]
I'm trying stuff out but things like this

`on enter of region:
	set {_pvprg} to "spawn-pvp-entry"
	set {_testrg} to "%region at player%"
	if {_testrg} contains {_pvprg}:
		message "Succes"`
or
`on enter of region:
	set {_pvprg} to "spawn-pvp-entry"
	if "%region at player%"contains {_pvprg}:
		message "Succes"`

Doesn't work either, and when I output both variables in the chat, I DO see the regions in both variables but it's not comparing them or something it seems?? If iI message them to me in chat, I see stuff like <region> in world world and stuff so thats good, but yeah..

**Comment by ShaneBeee** [comment]
> I'm trying stuff out but things like this
> 
> `on enter of region: set {_pvprg} to "spawn-pvp-entry" set {_testrg} to "%region at player%" if {_testrg} contains {_pvprg}: message "Succes"`
> or
> `on enter of region: set {_pvprg} to "spawn-pvp-entry" if "%region at player%"contains {_pvprg}: message "Succes"`
> 
> Doesn't work either, and when I output both variables in the chat, I DO see the regions in both variables but it's not comparing them or something it seems?? If iI message them to me in chat, I see stuff like in world world and stuff so thats good, but yeah..

Events are called before an event actually happens, so in this case, the event is called before the player actually enters the region, so at that point, the region at the player won't match. 
You can use event-region, or also (what I do) is wait 1 tick, and then compare with region at the player.


As per your loop in the OP, If im not mistaken this has never worked, the expression is:
`[(all|the)] blocks (in|of) [[the] region[s]] %regions%`
so `blocks in region "some.region"` won't work, because "some.region" is a string not a region object.
From my understanding this is not a bug, Skript has just never had a nice way of supporting regions.

Snow has made comments in the past about removing all region stuff from Skript and putting it in a separate addon. I'm not sure if/when this will happen, but I hope it does, because the region stuff is a royal pain in the behind due to WE/WG and their silly updates, not to mention FAWE using an old fork of WE/WG

**Comment by DarkObscurity** [comment]
So this basically means that for whatever reason it has always working, that has been fixed now so it won't work anymore? I used this alot... and I have no idea how to replace this functionality, because even if I put like 2 regions right before eachother so that the trigger is fired and right after that the player enters the region defined in the script, and that has always worked as well and if i like broadcast the regions it finds in the chat, I can see that it lists the region I try to use as a trigger, but it just doesn't do anything...

**Comment by EtoNikei** [comment]
Hello, what you say about calculating blocks in region? My shaft-regions contains 4050 blocks, but Skript returned the 5780
![image](https://user-images.githubusercontent.com/73131948/106590928-3be4c600-6580-11eb-9751-3b9c4cc21e06.png)

**Comment by AyhamAl-Ali** [comment]
> Hello, what you say about calculating blocks in region? My shaft-region contain 4050 blocks, but Skript returned the 5780 Skript 2.3.6, WorldGuard 6.2.2-SNAPSHOT;8eeab68 ![image](https://user-images.githubusercontent.com/73131948/106590928-3be4c600-6580-11eb-9751-3b9c4cc21e06.png)

This one should be fixed in #3708 report if it still exists

**Comment by AyhamAl-Ali** [comment]
Solved by this [comment](https://github.com/SkriptLang/Skript/issues/2737#issuecomment-570495362). If any ST think it should stay opened feel free to reopen.
